### PR TITLE
DB-10185 Add SYSCAT.REFERENCES view

### PR DESCRIPTION
**Pull request by ascend1**

### CODE REVIEW

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST FAILED
https://jenkins.build.splicemachine-dev.io/job/AutomatedTesting/job/spliceengine-PR/156/

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by jyuanca — APPROVED** [review]

File: splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyMetadataIT.java [hunk 1]
```diff
@@ -153,6 +157,44 @@ public void testSimplifiedForeignKeysDoesNotExplode() throws Exception{
...
+    public void testReferencesAndSysKeyColUseViews() throws Exception {
+        methodWatcher.executeUpdate("create table if not exists SELF_REF (c int, d int, " +
+                "constraint SRPK primary key (c), constraint SRFK foreign key (c) references SELF_REF(c))");
+
+        String query = "select * from syscat.references r where r.tabname = 'SELF_REF'";
+
+        String expected =
+                "CONSTNAME |      TABSCHEMA      | TABNAME | OWNER | OWNERTYPE |REFKEYNAME |    REFTABSCHEMA     |REFTABNAME |COLCOUNT |DELETERULE |UPDATERULE | CREATE_TIME |\n" +
+                "--------------------------------------------------------------------------------------------------------------------------------------------------------------\n" +
+                "   SRFK    |FOREIGNKEYMETADATAIT |SELF_REF | NULL  |   NULL    |   SRPK    |FOREIGNKEYMETADATAIT | SELF_REF  |    1    |     A     |     A     |    NULL     |";
```
Comment (hatyo) [inline splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyMetadataIT.java:170]: could you please add more tests for the following cases:

- foreign key with delete rules and update rules.
- table which is owned by a user (affecting `OWNER` table I guess)

**Comment by hatyo** [comment]
Great work @ascend1!

I would expand the commit message a bit with more information about this view if possible.

**Review by hatyo — CHANGES_REQUESTED** [review]
LGTM, but please consider adding more tests.

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyMetadataIT.java (hunk 1 above)
Comment (ascend1) [reply splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyMetadataIT.java]: Unused columns are removed and added test cases on different on delete/update rules.

**Review by hatyo — APPROVED** [review]

**Review by dgomezferro — APPROVED** [review]

**Review by arnaud-splice — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**